main.go

```diff
@@ -106,17 +80,31 @@ func boostrapper() pstore.PeerInfo {
 	}
 }
 
-func makeAndStartDHT(ds ds.Batching, addr string) (host.Host, *dht.IpfsDHT, error) {
-	h, err := makeBasicHost(addr)
+func makeAndStartNode(ds ds.Batching, addr string, relay bool) (host.Host, *dht.IpfsDHT, error) {
+	h, err := libp2p.New(context.Background(), libp2p.ListenAddrStrings(addr))
 	if err != nil {
 		panic(err)
 	}
 
+	if relay {
+		/* TODO: fix this, need @stebalien advice
```

Comment:
Pass `libp2p.EnableRelay(circuit.OptHop)` to the libp2p constructor.

Reply:
ah, super great